src/gui/gui_main_menu.cpp

```diff
@@ -297,6 +297,43 @@ void gamma_display::on_update(sys::state& state) noexcept {
 	set_text(state, "x" + text::format_float(state.user_settings.gamma));
 }
 
+void vassal_color_left::button_action(sys::state& state) noexcept {
+	auto index = uint8_t(state.user_settings.vassal_color);
+	if(state.user_settings.vassal_color > 0) {
+		state.user_settings.vassal_color = sys::map_vassal_color_mode(index - 1);
+		send(state, parent, notify_setting_update{});
+	}
+}
+void vassal_color_left::on_update(sys::state& state) noexcept {
+	disabled = (state.user_settings.vassal_color == sys::map_vassal_color_mode(0));
+}
+void vassal_color_right::button_action(sys::state& state) noexcept {
+	auto index = uint8_t(state.user_settings.vassal_color);
+	if(state.user_settings.vassal_color < sys::map_vassal_color_mode(2)) {
+		state.user_settings.vassal_color = sys::map_vassal_color_mode(index + 1);
```

Comment:
this won't work, visually, because it doesn't actually force the colors to be regenerated